src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -149,6 +149,76 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_win_set_per_win_sync(MPIR_Win * win)
     goto fn_exit;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIDI_OFI_win_init_sep
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+/* Activate per-window EP/counter using scalable ep. */
+MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_win_init_sep(MPIR_Win * win)
+{
+    int i, mpi_errno = MPI_SUCCESS;
+    struct fi_info *finfo;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_WIN_INIT_SEP);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_WIN_INIT_SEP);
+
+    finfo = fi_dupinfo(MPIDI_Global.prov_use);
+    MPIR_Assert(finfo);
+
+    /* Initialize scalable EP when first window is created. */
+    if (MPIDI_Global.rma_sep == NULL) {
+        /* Specify number of transmit context according user input and provider limit. */
+        MPIDI_Global.max_rma_tx_cnt = MPL_MIN(MPIDI_Global.prov_use->domain_attr->max_ep_tx_ctx,
+                                              MPIR_CVAR_CH4_OFI_MAX_RMA_SEP_CTX);
+        finfo->ep_attr->tx_ctx_cnt = MPIDI_Global.max_rma_tx_cnt;
```

Comment:
It looks like the number of available transmit contexts in the scalable EP is only `1` by default. Can we set the default value of `MPIR_CVAR_CH4_OFI_MAX_RMA_SEP_CTX ` to something like "auto", thus `prov_use->domain_attr->max_ep_tx_ctx` number of transmit contexts can be used ?

Reply:
By default, current RMA path only uses `1` transmit context per window. So I just don't want to exhaust system resource by using `prov_use->domain_attr->max_ep_tx_ctx`.

Reply:
The scalable EP is shared by all windows, isn't it ? 
For example, only the first window can get a transmit context, then the second window will fail at `utarray_back(MPIDI_Global.rma_sep_idx_array)` . Please let me know if my understanding is wrong.

Reply:
Correct. Currently, `rma` and `pt2pt` are in the same domain. Take an example of `fi_info` in `psm2`:
```
        tx_ctx_cnt: 125
        rx_ctx_cnt: 125
        max_ep_tx_ctx: 128
        max_ep_rx_ctx: 128
```
And in `sockets`:
```
        tx_ctx_cnt: 16
        rx_ctx_cnt: 16
        max_ep_tx_ctx: 16
        max_ep_rx_ctx: 16
```
Making it as `1` is not good idea for window2, but set as maximum value exhausts all tx in the same domain.

Reply:
If set `finfo->ep_attr->tx_ctx_cnt= max_ep_tx_ctx ` , are all `max_ep_tx_ctx ` number of contexts reserved at the SEP creation time ? 

In current design, by default, the first window will create the global SEP with 1 `tx_ctx`, and then the second window will create the global shared context and share it with all later windows. This is misleading and the SEP with single `tx_ctx` is not useful here.

Since the SEP option is not useful unless the user specifies `MPIR_CVAR_CH4_OFI_MAX_RMA_SEP_CTX >1`, can we just disable SEP by default ?

Reply:
For first question, I think yes (as far as I can tell is from libfabric man page).

My original thought is to test `tx` in `sep` can work for RMA. It is hard to tell how many tx per sep we should set at this moment.  If you have multiple windows, then num of tx should be larger than number of window to be consistent.

I can turn sep off by default.

@hajimefu Any comments?

Reply:
@shawnccx I'm okay with turning it off by default, as long as it can be enabled via a CVAR.